src/bql.py

```diff
@@ -703,6 +727,20 @@ def _create_population(bdb, phrase):
     else:
         pop_guess_vars = []
 
+    # Ensure no string-valued variables are being modeled as numerical.
+    numerical_string_vars = []
+    for var, stattype in pop_model_vars:
+        if stattype == 'numerical':
+            qt = sqlite3_quote_name(phrase.table)
+            cursor = bdb.sql_execute('SELECT %s FROM %s' % (var, qt))
+            rows = cursor.fetchall()
+            if any(isinstance(r[0], (unicode, str)) for r in rows):
+                numerical_string_vars.append(var)
+    if len(numerical_string_vars) > 0:
+        raise BQLError(
+            bdb, 'Column(s) with string values modeled as numerical: %r'
+            % (numerical_string_vars, ))
+
```

Comment:
Code like this should almost never be duplicated verbatim. If there is a need for repeated functionality, then wrap the functionality inside a function.